lib/travis/services/find_admin.rb

```diff
@@ -29,25 +29,33 @@ def repository
       private
 
         def candidates
-          User.with_github_token.with_permissions(:repository_id => repository.id, :admin => true)
+          User.with_github_token.with_permissions(:repository_id => repository.id, :admin => true).first(NUM_CANDIDATES)
+        end
+
+        def admin
+          admin = candidates.detect do |candidate|
+            is_valid? candidate
+          end
+          admin || raise_admin_missing
         end
 
-        def validate(user)
+        def is_valid?(admin)
           Timeout.timeout(2) do
-            data = Github.authenticated(user) { repository_data }
+            data = Github.authenticated(admin) { repository_data }
             if data['permissions'] && data['permissions']['admin']
-              user
+              true
             else
-              info "[github-admin] #{user.login} no longer has admin access to #{repository.slug}"
-              update(user, data['permissions'])
+              info "[github-admin] #{admin.login} no longer has admin access to #{repository.slug}"
+
+              admin.update_attributes!(:permissions => { :admin => false })
```

Comment:
I'm not sure what ActiveRecord does in this case with a `has_many :through` relationship. I'd suspect that, if it does anything at all (might raise?), it would still unset all permissions for that user, and I'd guess that's not what you intend here.

How about doing something like this instead: 

```
Permission.where(user_id: user.id, repository_id: repository.id).update_all(admin: false)
```